packages/node_modules/@ciscospark/widget-recents/src/container.js

```diff
@@ -140,51 +148,49 @@ export class RecentsWidget extends Component {
       user,
       spaces,
       spacesList,
-      sparkInstance,
-      widgetRecents
+      sparkInstance
     } = nextProps;
 
     if (user.get(`currentUser`)
       && spaces.getIn(`status`, `hasFetched`)
       && this.props.spacesList
       && spacesList.count() > this.props.spacesList.count()
       ) {
-      if (!widgetRecents.getIn([`status`, `isListeningForNewActivity`])) {
-        this.listenForNewActivity(sparkInstance, nextProps);
-        this.listenForCall(sparkInstance, nextProps);
-      }
-      spaces.get(`items`).forEach((s) => {
-        if (s.get(`type`) === `direct`) {
-          if (s.get(`participants`).count() > 1) {
-            // Get the user ID of the participant that isn't current user
-            const currentUserId = user.get(`currentUser`).id;
-            const otherUser = s.get(`participants`).find((p) => p.get(`id`) !== currentUserId);
-            if (otherUser) {
-              const otherUserId = otherUser.get(`id`);
-              this.fetchAvatar({userId: otherUserId}, nextProps);
-            }
-          }
+      spacesList.forEach((s) => {
+        if (s.type === `direct` && s.participants.length > 1 && s.toPersonId) {
```

Comment:
Are the spaces in `spacesList` an immutable object? Would you need to `get` those props?

Reply:
So I'll probably end up changing this in the future, but spacesList itself is immutable, but it's an ordered list of plain objects, so this works right now. Should probably go one way or the other (thinking just a plain object)

Reply:
That's why I've just been converting them all in the selector to regular js objects.

Reply:
With this it's a bit different because the order matters, so OrderedMap is the best way to do this and plain js objects wouldn't maintain order while also allowing us to retrieve space data by ID